src/__tests__/ceramic_integration.test.ts

```diff
@@ -155,6 +156,24 @@ async function waitForAnchor(stream: Stream): Promise<void> {
     .toPromise()
 }
 
+async function waitForUpdate(stream: Stream, update: CID, timeoutMS = 30 * 1000): Promise<void> {
+  await firstValueFrom(
+    stream.pipe(
+      filter((state) => {
+        return state.log[state.log.length - 1].cid.toString() === update.toString()
+      }),
+      timeout({
```

Comment:
does the `timeout` have to be first in the `pipe` to be processed?  I don't know enough about how RXJS works, but I'd have thought the `pipe` processes each entry one-at-a-time and so the `timeout` wouldn't be processed until after the `filter` resolved?  Maybe not though - did you test the timeout behavior?  If you did then it's fine

Reply:
I can definitely see how you would think that. From the docs `timeout` errors if Observable does not emit a value in given time span. So I think it has to come after the filter. I have tested this works.